src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -0,0 +1,435 @@
+import de.hpi.bpt.argos.common.EventProcessingPlatformUpdaterImpl;
+import de.hpi.bpt.argos.storage.PersistenceAdapterImpl;
+import de.hpi.bpt.argos.storage.dataModel.attribute.type.TypeAttribute;
+import de.hpi.bpt.argos.storage.dataModel.attribute.type.TypeAttributeImpl;
+import de.hpi.bpt.argos.storage.dataModel.event.Event;
+import de.hpi.bpt.argos.storage.dataModel.event.query.EventQuery;
+import de.hpi.bpt.argos.storage.dataModel.event.type.EventType;
+import de.hpi.bpt.argos.storage.dataModel.event.type.EventTypeImpl;
+import de.hpi.bpt.argos.storage.dataModel.mapping.EventEntityMapping;
+import de.hpi.bpt.argos.storage.dataModel.mapping.MappingCondition;
+import de.hpi.bpt.argos.util.HttpStatusCodes;
+import de.hpi.bpt.argos.util.RestEndpointUtil;
+import de.hpi.bpt.argos.util.RestEndpointUtilImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.HaltException;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+import static spark.Spark.halt;
+
+import java.util.List;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EventTypeEndpointImpl implements EventTypeEndpoint {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeEndpointImpl.class);
+    private static final Gson serializer = new Gson();
+    private static final RestEndpointUtil endpointUtil = RestEndpointUtilImpl.getInstance();
+    private static final JsonParser jsonParser = new JsonParser();
+
+    private static final String JSON_EVENT_TYPE_ATTRIBUTE = "EventType";
+    private static final String JSON_NAME_ATTRIBUTE = "Name";
+    private static final String JSON_ATTRIBUTES_ATTRIBUTE = "TypeAttributes";
+    private static final String JSON_TIMESTAMP_ATTRIBUTE = "TimeStampAttributeName";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.get(EventTypeEndpoint.getEventTypesBaseUri(), this::getEventType);
+        sparkService.get(EventTypeEndpoint.getEventTypeBaseUri(), this::getEventTypes);
+        sparkService.post(EventTypeEndpoint.getCreateEventTypeBaseUri(), this::createEventType);
+        sparkService.delete(EventTypeEndpoint.getDeleteEventTypeBaseUri(), this::deleteEventType);
+        sparkService.get(EventTypeEndpoint.getEventTypeAttributesBaseUri(), this::getEventTypeAttributes);
+        sparkService.get(EventTypeEndpoint.getEventTypeQueriesBaseUri(), this::getEventTypeQueries);
+        sparkService.get(EventTypeEndpoint.getEventTypeEntityMappingsBaseUri(), this::getEventTypeEntityMappings);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEventTypes(Request request, Response response) {
+        List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+        JsonArray jsonEventTypes = new JsonArray();
+        // TODO where does the key for each event type come from? ("EventType{...}")

```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1135)